pkgs/os-specific/linux/ryzen-smu/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitLab, kernel, linuxHeaders }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ryzen_smu-${kernel.version}";
+  version = "0.1.5";
+
+  src = fetchFromGitLab {
+    owner = "mann1x";
```

Comment:
The ryzen_smu programs have gone through a lot of forks themselves.

[ryzen_smu](https://gitlab.com/leogx9r/ryzen_smu) is the original ryzen_smu. It included `monitor_cpu`. [ryzen_monitor](https://github.com/hattedsquirrel/ryzen_monitor) extended `monitor_cpu` and [ryzen_monitor_ng](https://github.com/mann1x/ryzen_monitor_ng) extended `ryzen_monitor`.

To support [ryzen_monitor_ng](https://github.com/mann1x/ryzen_monitor_ng), `mann1x` forked `ryzen_smu` but hasn't merged the changes. See [here](https://github.com/mann1x/ryzen_monitor_ng#building) and [here](https://gitlab.com/leogx9r/ryzen_smu/-/issues/26#note_1666173169).

However, when I built `ryzen_monitor_ng` from [mann1x's repo](https://github.com/mann1x/ryzen_monitor_ng) it wouldn't run because of a version mismatch in `libsmu`. [kvic-z's fork of ryzen_monitor_ng](https://github.com/kvic-z/ryzen_monitor_ng) corrects the version mismatch and adds Matisse.

The patches you suggest come from yet another fork.

I do like your approach of patching the original repos. I hope it doesn't lead to conflicts between `mann1x's` fork and `moson-mo's`. Worth a try, even if it leads to a lot of patches.

Reply:
It's so complex that I have just realized. I have noticed that the origin repo hasn't updated since half year ago. Do you think if it's a good idea that we maintain a fork in person, picking PRs and testing, then releasing.

Anyway, please at least leave a comment in the Nix packaging source code about it as long as you choose a repo which is not the origin one. Otherwise, other people who don't see this discussion would get confused.

Reply:
It is looking like we will need to maintain a repo with all the patches. PRs are not getting merged upstream quickly. Are you willing to set up the repo? I will change the `src` if you do.

Reply:
> To support [ryzen_monitor_ng](https://github.com/mann1x/ryzen_monitor_ng), `mann1x` forked `ryzen_smu` but hasn't merged the changes. See [here](https://github.com/mann1x/ryzen_monitor_ng#building) and [here](https://gitlab.com/leogx9r/ryzen_smu/-/issues/26#note_1666173169).

I have noticed the same commits with different sha256 in mann1x's repo and leogx9r's repo, such as https://gitlab.com/leogx9r/ryzen_smu/-/commit/e61177d0ddaebfaeca52094b20a2289287a0838b and https://gitlab.com/mann1x/ryzen_smu/-/commit/adaf53c77dc68b99643ab7af729eb9c592cf5e01. Has it already rebased or I  miss something other place.